src/aggregation/bucket/range.rs

```diff
@@ -321,56 +345,75 @@ fn to_u64_range(range: &RangeAggregationRange, field_type: &Type) -> crate::Resu
         u64::MAX
     };
 
-    Ok(start..end)
+    Ok(InternalRangeAggregationRange {
+        key: range.key.clone(),
+        range: start..end,
+    })
 }
 
 /// Extends the provided buckets to contain the whole value range, by inserting buckets at the
 /// beginning and end and filling gaps.
 fn extend_validate_ranges(
     buckets: &[RangeAggregationRange],
     field_type: &Type,
-) -> crate::Result<Vec<Range<u64>>> {
+) -> crate::Result<Vec<InternalRangeAggregationRange>> {
     let mut converted_buckets = buckets
         .iter()
         .map(|range| to_u64_range(range, field_type))
         .collect::<crate::Result<Vec<_>>>()?;
 
-    converted_buckets.sort_by_key(|bucket| bucket.start);
-    if converted_buckets[0].start != u64::MIN {
-        converted_buckets.insert(0, u64::MIN..converted_buckets[0].start);
+    converted_buckets.sort_by_key(|bucket| bucket.range.start);
+    if converted_buckets[0].range.start != u64::MIN {
+        converted_buckets.insert(
+            0,
+            InternalRangeAggregationRange {
+                key: None,
+                range: u64::MIN..converted_buckets[0].range.start,
+            },
```

Comment:
I would add a 
`impl From<Range<u64>> for InternalRangeAggregationRange` and then use into

Reply:
Oh I see,  refactored so!
https://github.com/quickwit-oss/tantivy/pull/1426/commits/099e62615624e5cc0c34302f25670a5e8c05ce0c